### PR TITLE
fix: replace calls to deprecated

### LINKED ISSUE #350: Numpy 2.0 compatibility
There are some cases when plopp uses old numpy functionality.

Example:
```
 File ~/work/esssans/esssans/.tox/docs/lib/python3.10/site-packages/plopp/graphics/colormapper.py:113, in ColorMapper.__init__(self, canvas, cbar, cmap, mask_cmap, norm, autoscale, vmin, vmax, nan_color, figsize)
    111 self.user_vmax = vmax
    112 self.vmin = np.inf
--> 113 self.vmax = np.NINF
    114 self.norm = norm
    115 self._autoscale = autoscale

 AttributeError: `np.NINF` was removed in the NumPy 2.0 release. Use `-np.inf` instead.
```

### PR DESCRIPTION
**Pull request by jokasimr**
Fixes #350

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -9,18 +9,18 @@ contourpy==1.2.1
...
     # via matplotlib
-fonttools==4.51.0
+fonttools==4.53.0
     # via matplotlib
 kiwisolver==1.4.5
     # via matplotlib
 matplotlib==3.9.0
     # via -r base.in
-numpy==1.26.4
+numpy==2.0.0
```
Comment (nvaytet) [inline requirements/base.txt:18]: Is `scipp` compatible now with numpy 2.0? I thought we had not yet released that?

Can we make the changes with `np.NINF` etc without updating the dependencies?

File: requirements/nightly.in [hunk 2]
```diff
@@ -3,4 +3,4 @@
 # --- END OF CUSTOM SECTION ---
 # The following was generated by 'tox -e deps', DO NOT EDIT MANUALLY!
 matplotlib>=3.5
-https://github.com/scipp/scipp/releases/download/nightly/scipp-nightly-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+https://github.com/scipp/scipp/releases/download/nightly/scipp-nightly-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```
Comment (nvaytet) [inline requirements/nightly.in:6]: I don't think we have dropped 3.10 yet?

Comment (jokasimr) [reply requirements/nightly.in:6]: Not sure why this changed. I just ran `tox -e deps` :confused:

**Review by nvaytet — APPROVED** [review]

File: requirements/nightly.in (hunk 2 above)
Comment (MridulS) [reply requirements/nightly.in]: Yeah, it picks up the local env python and forces it through https://github.com/scipp/copier_template/issues/163

Comment (jokasimr) [reply requirements/nightly.in]: Good to know, thanks!

**Pull request merged**